### PR TITLE
Include the reason for AbstractClient connection failure

### PR DESCRIPTION
**Pull request by ggezer**

### CODE REVIEW

**Comment by ggezer** [comment]
@cheng-chang PTAL.

**Comment by cheng-chang** [comment]
alluxio-bot, merge this please

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Comment by ggezer** [comment]
@calvinjia PTAL

File: core/common/src/main/java/alluxio/AbstractClient.java [hunk 1]
```diff
@@ -259,12 +257,15 @@ public synchronized void connect() throws AlluxioStatusException {
...
+    /**
+     * Throw as-is if {@link UnauthenticatedException} occurred.
+     */
+    if (lastConnectFailure instanceof UnauthenticatedException) {
+      throw (AlluxioStatusException) lastConnectFailure;
     }
 
     throw new UnavailableException(String.format("Failed to connect to %s @ %s after %s attempts",
-        getServiceName(), mAddress, retryPolicy.getAttemptCount()));
+        getServiceName(), mAddress, retryPolicy.getAttemptCount()), lastConnectFailure);
```
Comment (calvinjia) [inline core/common/src/main/java/alluxio/AbstractClient.java]: Is last connect failure guaranteed to be non null?

Comment (ggezer) [reply core/common/src/main/java/alluxio/AbstractClient.java]: No. However it's allowed to have null in the places it has been used.

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Pull request merged**